tests/unit_tests/CMakeLists.txt

```diff
@@ -182,6 +182,9 @@ foreach(domain ${TARGET_DOMAINS})
       PROPERTIES TEST_PREFIX ${DOMAIN_PREFIX}/RT/
       DISCOVERY_TIMEOUT 30
     )
+
+    # ensure rt tests have up-to-date libs
+    add_dependencies(test_main_${domain}_rt ${ONEMKL_LIBRARIES_${domain}})
```

Comment:
In most cases `add_dependencies` shouldn't be needed if we use `target_link_libraries`. I can see the compile tests above use `add_dependencies` but I'm not sure why, do you know if there is a good reason for it?